packages/pyright-internal/src/analyzer/typeEvaluator.ts

```diff
@@ -14625,6 +14631,13 @@ export function createTypeEvaluator(importLookup: ImportLookup, evaluatorOptions
             return undefined;
         }
 
+        // Very complex functions with many arguments can take a long time to analyze,
+        // so we'll use a heuristic and avoiding this inference technique for any
+        // call site that involves too many arguments.
+        if (args.length > maxReturnTypeInferenceArgumentCount) {
+            return undefined;
+        }
```

Comment:
I was also thinking about counting the unknown parameters when calculating allArgTypesAreUnknown and if the number of unknown parameters was over a certain value dont evaluate inferFunctionReturnType

Reply:
No your fix is better

Reply:
testing this again.